### PR TITLE
[recipe, doc] feat: add mixgrpo/grpoguard and full-weight flowgrpo fo…

### PR DESCRIPTION
**Pull request by wsc16**
…r Qwen-Image with npu bakcend

### What does this PR do?

Add npu training scripts for Qwen-Image model, including: 1. mixgrpo, 2. grpoguard, 3. full-weight Qwen-Image flowgrpo.

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `vllm_omni`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `diffusion`, `omni`, `tests`, `docker`
  - If this PR involves multiple modules, separate them with `,` like `[diffusion, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][diffusion, fsdp] feat: new rollout scheduler`

### Test
1. result of mixgrpo validated on Atlas 800T A2 (CANN 9.0.0)
The reward value at step 0, 30, 60, 90 are 0.8999, 0.9074, 0.9306, and 0.9323, and the loss and reward curves are as follow:

actor
image

critic
image

2. result of grpoguard validated on Atlas 800T A2 (CANN 9.0.0)
The reward value at step 0, 30, 60, 90 are 0.9013, 0.9173, 0.9524, and the loss and reward curves are as follow:

actor
image

critic
image

val
image

3. result of full-weight flowgrpo validated  on Atlas 800T A3 (CANN 9.0.0) with 16 cards
The reward value at step 30, 60, 90, 120 are 0.5186, 0.6985, 0.88, and 0.9232, and the loss and reward curves are as follow:

actor
image

critic
image

val
image

### API and Usage Example
```bash
bash examples/mixgrpo_trainer/run_qwen_image_ocr_lora_mixgrpo_npu.sh
```
```bash
bash examples/grpoguard_trainer/run_qwen_image_ocr_lora_npu.sh
```
```bash
bash examples/flowgrpo_trainer/run_qwen_image_ocr_npu.sh
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/verl-project/verl-omni/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/verl-project/verl-omni/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [x] Add / Update the documentation.
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl-omni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

File: examples/flowgrpo_trainer/README.md [hunk 1]
```diff
@@ -104,6 +104,11 @@ We have provided a script to enable non-cfg full-weight Qwen-Image OCR training.
 bash examples/flowgrpo_trainer/run_qwen_image_ocr.sh
 ```
 
+An NPU script for Atlas A3 with 16 NPUs is also provided. Before running, set the `ASCEND_HOME_PATH` environment variable (defaults to `/usr/local/Ascend/cann-9.0.0`).
```
Comment (panshaowu) [inline examples/flowgrpo_trainer/README.md:107]: Since the environment variable `ASCEND_HOME_PATH` has already been set in the script, is it still necessary to explain it here?

File: examples/grpoguard_trainer/README.md [hunk 2]
```diff
@@ -58,7 +68,16 @@ The script runs `python3 -m verl_omni.trainer.main_diffusion` with:
 - `actor_rollout_ref.actor.diffusion_loss.clip_ratio=2e-6`
 - `actor_rollout_ref.rollout.algo.sde_type=sde`
 - `reward.custom_reward_function.name=compute_score_ocr`
-- `trainer.n_gpus_per_node=4`
+
+Due to memory differences, the NPU and GPU configurations differ as follows:
```
Comment (panshaowu) [inline examples/grpoguard_trainer/README.md]: It is recommended to replace "memory differences" with "differences in memory capacity".

File: examples/flowgrpo_trainer/README.md (hunk 1 above)
Comment (wsc16) [reply examples/flowgrpo_trainer/README.md:107]: ASCEND_HOME_PATH is set with a default value, but users may use a custom installation path of CANN.

File: docs/start/install.md [hunk 3]
```diff
@@ -28,7 +28,7 @@ source .venv/bin/activate
 For NVIDIA GPU:
 
 ```bash
-uv pip install vllm==0.20.2
+uv pip install "vllm @ git+https://github.com/vllm-project/vllm.git@releases/v0.20.2"
```
Comment (AndyZhou952) [inline docs/start/install.md]: I don't understand the edit here - any specific reason to clone from GitHub?

Comment (wsc16) [reply docs/start/install.md]: The vLLM needs to be installed using the source code on NPU. The code location modified ealier is incorrect.

**Comment by zhtmike** [comment]
Ready for merge? @wsc16 @panshaowu @AndyZhou952

File: docs/start/install.md
Comment (AndyZhou952) [inline docs/start/install.md]: Please also update the date here.

Comment (wsc16) [reply docs/start/install.md]: done

**Review by AndyZhou952 — APPROVED** [review]

**Review by zhtmike — APPROVED** [review]
